Mathlib/Topology/Algebra/Category/ProfiniteGrp/Basic.lean

```diff
@@ -161,4 +161,101 @@ instance : HasForget₂ ProfiniteGrp Grp where
     map := fun f => f.toMonoidHom
   }
 
+/--The functor mapping a profinite group to its underlying profinite space-/
+def profiniteGrpToProfinite : ProfiniteGrp ⥤ Profinite where
+  obj G := G.toProfinite
+  map f := ⟨f, by continuity⟩
+
+instance : profiniteGrpToProfinite.Faithful := {
+  map_injective := fun {_ _} _ _ h =>
+    ConcreteCategory.hom_ext_iff.mpr (congrFun (congrArg ContinuousMap.toFun h)) }
+
+end ProfiniteGrp
+
+/-!
+# Limits in the category of profinite groups
+
+* `ProfiniteGrp.limitCone` : The explicit limit cone in `ProfiniteGrp`.
+
+* `ProfiniteGrp.limitConeIsLimit`: `ProfiniteGrp.limitCone` is a limit cone.
+
+-/
+
+section Limits
+
+/- In this section, we construct limits in the category of profinite groups. -/
+
+namespace ProfiniteGrp
+
+section
+
+variable {J : Type v} [SmallCategory J] (F : J ⥤ ProfiniteGrp.{max v u})
+
+/-- Auxiliary construction to obtain the group structure on the limit of profinite groups. -/
+def limitConePtAux : Subgroup (Π j : J, F.obj j) where
+  carrier := {x | ∀ ⦃i j : J⦄ (π : i ⟶ j), F.map π (x i) = x j}
+  mul_mem' hx hy _ _ π := by simp only [Pi.mul_apply, map_mul, hx π, hy π]
+  one_mem' := by simp only [Set.mem_setOf_eq, Pi.one_apply, map_one, implies_true]
+  inv_mem' h _ _ π := by simp only [Pi.inv_apply, map_inv, h π]
+
+instance : Group (Profinite.limitCone (F ⋙ profiniteGrpToProfinite.{max v u})).pt :=
+  inferInstanceAs (Group (limitConePtAux F))
+
+instance : TopologicalGroup (Profinite.limitCone (F ⋙ profiniteGrpToProfinite.{max v u})).pt :=
+  inferInstanceAs (TopologicalGroup (limitConePtAux F))
+
+/-- The explicit limit cone in `ProfiniteGrp`. -/
+def limitCone : Limits.Cone F where
+  pt := ofProfinite (Profinite.limitCone (F ⋙ profiniteGrpToProfinite.{max v u})).pt
+  π :=
+  { app := fun j => {
+      toFun := fun x => x.1 j
+      map_one' := rfl
+      map_mul' := fun x y => rfl
+      continuous_toFun := by
+        exact (continuous_apply j).comp (continuous_iff_le_induced.mpr fun U a => a) }
+    naturality := fun i j f => by
+      simp only [Functor.const_obj_obj, Functor.comp_obj,
+        Functor.const_obj_map, Category.id_comp, Functor.comp_map]
+      congr
+      exact funext fun x => (x.2 f).symm }
+
+/-- `ProfiniteGrp.limitCone` is a limit cone. -/
+def limitConeIsLimit : Limits.IsLimit (limitCone F) where
+  lift cone := {
+    toFun := ((Profinite.limitConeIsLimit (F ⋙ profiniteGrpToProfinite)).lift
+      (profiniteGrpToProfinite.mapCone cone)).toFun
+    map_one' := by
+      apply SetCoe.ext
```

Comment:
I'm a bit sad to see that this is necessary (rather than just calling `ext`). It indicates to me that perhaps we haven't set up `@[ext]` lemmas sufficiently for `TopCat`.

Doesn't need to block this PR, I'll pass that decision back to @dagurtomas. :-)

Reply:
Let me investigate a bit

Reply:
I don't have time to dig more so I'd recommend that we just merge this. I added a TODO. 

Let me just note that even adding the lemma
```lean 
@[ext]
lemma limitCone_ext {J : Type v} [SmallCategory J] {F : J ⥤ ProfiniteGrp.{max v u}}
    {x y : (limitCone F).pt} (h : ∀ i, x.1 i = y.1 i) : x = y := Subtype.ext (funext fun i ↦ h i)
```
wasn't enough to get the `ext` tactic to work.